### PR TITLE
Added Special Character Support

### PR DESCRIPTION
**Pull request by 16bitmood**
Special character support for textbox2d (like: '!@#$%^&*()')

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @16bitmood,

Thank you for adding this feature! we will look into it and get back to you.

File: fury/ui.py [hunk 1]
```diff
@@ -1896,6 +1901,7 @@ class LineSlider2D(UI):
     active_color : (float, float, float)
         Color of the handle when it is pressed.
     """
+
     def __init__(self, center=(0, 0),
                  initial_value=50, min_value=0, max_value=100,
                  length=200, line_width=5,
```
Comment (Nibba2018) [inline fury/ui.py:1907]: Are these extra new lines necessary?

File: fury/ui.py [hunk 2]
```diff
@@ -2244,6 +2250,7 @@ class LineDoubleSlider2D(UI):
         Color of the handles when they are pressed.
 
     """
+
     def __init__(self, line_width=5, inner_radius=0, outer_radius=10,
                  handle_side=20, center=(450, 300), length=200,
```
Comment (Nibba2018) [inline fury/ui.py:2255]: same as above and below few new lines.

**Review by Nibba2018 — CHANGES_REQUESTED** [review]
Hi @16bitmood , thanks for creating this PR, I have tested various special characters and they work fine on `TextBox2D`.

But I don't understand your unit test. Please be a bit more elaborate. For me a window pops up and then quickly closes without any keystrokes.

I would suggest you to add some assertions at the end to test whether the key presses are correctly registered/saved or not. For e.g:

* type a string including printable characters.
* close the window.
* then assert if the message parameter contains the same string of the characters or not.

Thanks!

**Pull request closed without merging**